Use MSVC 2017 compatible syntax

wada314 opened the pull request:
Because we are using designated initializer feature at line 47 & 51 (after this patch), this code does not compile in VS2017, and the error message for that is not clear.

According to MS dev blog, this feature is introduced in VS 2019 16.1.
https://devblogs.microsoft.com/cppblog/c20-features-and-fixes-in-vs-2019-16-1-through-16-6/

twistedfall commented:
Thanks for the pull request! Do you happen to know if there are any other features used in the project that require VS 2019? Because if not, I would fix the code to not require such a new compiler.

wada314 commented:
Hello! 😄 
I confirmed that if you fix those 2 places then the code can be compiled with VS2017.
// I tried `cargo build` in the project's root directory and it successfully finished.

twistedfall commented:
Thanks for checking! Can you then replace the changes in this PR with those fixes? I think it's the best way forward.

wada314 committed:
Use MSVC 2017 compatible syntax

Unfortunately C++ struct's designated initializer syntax is not supported in VS 2017 compiler,
so use the traditional initializer syntax.
https://devblogs.microsoft.com/cppblog/c20-features-and-fixes-in-vs-2019-16-1-through-16-6/

wada314 commented:
Sure! Updated the PR to use the traditional syntax.

twistedfall commented:
Thanks a lot!
